homeassistant/components/shelly/climate.py

```diff
@@ -0,0 +1,201 @@
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.components.shelly import BlockDeviceWrapper
+from homeassistant.components.shelly.entity import ShellyBlockEntity
+from homeassistant.components.shelly.utils import get_device_entry_gen
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import (
+    AIOSHELLY_DEVICE_TIMEOUT_SEC,
+    BLOCK,
+    DATA_CONFIG_ENTRY,
+    DOMAIN,
+    SHTRV_01_TEMPERATURE_SETTINGS,
+)
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up climate device."""
+
+    if get_device_entry_gen(config_entry) == 2:
+        return
+
+    wrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id][BLOCK]
+    for block in wrapper.device.blocks:
+        if hasattr(block, "targetTemp"):
+            async_add_entities([ShellyClimate(wrapper, block)])
+            break
+
+
+class ShellyClimate(ShellyBlockEntity, RestoreEntity, ClimateEntity):
+    """Representation of a Shelly climate device."""
+
+    _attr_hvac_modes = [HVAC_MODE_OFF, HVAC_MODE_HEAT]
+    _attr_icon = "mdi:thermostat"
+    _attr_max_temp = SHTRV_01_TEMPERATURE_SETTINGS["max"]
+    _attr_min_temp = SHTRV_01_TEMPERATURE_SETTINGS["min"]
+    _attr_supported_features: int = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+    _attr_target_temperature_step = SHTRV_01_TEMPERATURE_SETTINGS["step"]
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, wrapper: BlockDeviceWrapper, block: Block) -> None:
+        """Initialize climate."""
+        super().__init__(wrapper, block)
+
+        self.wrapper = wrapper
+        self.block = block
+
+        assert self.block.channel
+
+        self._load_device_block()
+
+        self.control_result: dict[str, Any] | None = None
+
+        self._attr_name = self.wrapper.name
+        self._attr_unique_id = self.wrapper.mac
+        self._attr_preset_modes: list[str] = [
+            PRESET_NONE,
+            *wrapper.device.settings["thermostats"][int(self.block.channel)][
+                "schedule_profile_names"
+            ],
+        ]
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Set target temperature."""
+        return cast(float, self.block.targetTemp)
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return current temperature."""
+        return cast(float, self.block.temp)
+
+    @property
+    def available(self) -> bool:
+        """Device availability."""
+        return not cast(bool, self._device_block.valveError)
```

Comment:
```suggestion
        return not self._device_block.valveError
```

I think you can drop the `cast` here

Reply:
Seems not, don't know why but without `cast` is always unavailable